web/server/codechecker_server/migrations/report/versions/75ae226b5d88_review_status_for_each_report.py

```diff
@@ -0,0 +1,177 @@
+        """)
+
+    conn.execute("""
+        UPDATE reports
+        SET review_status = 'unreviewed'
+        WHERE review_status IS NULL
+    """)
+
+    files_with_report = conn.execute("""
+        SELECT DISTINCT reports.file_id, files.content_hash
+        FROM reports INNER JOIN files ON reports.file_id = files.id
+        WHERE review_status != 'unreviewed'
+    """)
+
+    content_hashes = set()
+    hash_to_content = {}
+    file_id_to_content_hash = {}
+
+    for f in files_with_report:
+        content_hashes.add(f"'{f.content_hash}'")
+        file_id_to_content_hash[f.file_id] = f.content_hash
+
+    if content_hashes:
+        hash_to_content = conn.execute(f"""
+            SELECT content_hash, content FROM file_contents
+            WHERE content_hash IN ({','.join(content_hashes)})
+        """)
+
+    hash_to_content = {
+        x.content_hash: decode_file_content(x.content)
+        for x in hash_to_content}
+
+    report_id_to_line = conn.execute(f"""
+        SELECT id, file_id, bug_id, checker_id, line FROM reports
+        WHERE review_status != 'unreviewed'
+    """)
+
+    scch = SourceCodeCommentHandler()
+    comment_cache = {}
+    bug_hashes = set()
+    review_status_in_source = set()
+
+    for row in report_id_to_line:
+        cache_key = (file_id_to_content_hash[row.file_id], row.line)
+        if cache_key not in comment_cache:
+            try:
+                comment_cache[cache_key] = scch.get_source_line_comments(
+                    StringIO(
+                        hash_to_content[file_id_to_content_hash[row.file_id]]),
+                    row.line)
+            except SpellException:
+                comment_cache[cache_key] = []
+
+        for comment in comment_cache[cache_key]:
+            if row.checker_id in comment.checkers or \
+                    'all' in comment.checkers:
+                review_status_in_source.add(row.id)
+                bug_hashes.add(f"'{row.bug_id}'")
+
+    if review_status_in_source:
+        conn.execute(f"""
+            UPDATE reports
+            SET review_status_is_in_source = '1'
+            WHERE id IN ({','.join(map(str, review_status_in_source))})
+        """)
+
+    results = conn.execute("""
+        SELECT bug_hash, date
+        FROM review_statuses
+        WHERE status IN ('false_positive', 'intentional')
+    """)
+
+    for row in results:
+        if dialect == 'sqlite':
+            conn.execute(f"""
+                UPDATE reports
+                SET fixed_at = max('{row.date}', detected_at)
+                WHERE fixed_at IS NULL AND bug_id = '{row.bug_hash}'
+            """)
+        elif dialect == 'postgresql':
+            conn.execute(f"""
+                UPDATE reports
+                SET fixed_at = greatest('{row.date}', detected_at)::timestamp
+                WHERE fixed_at IS NULL AND bug_id = '{row.bug_hash}'
+            """)
+
+    if bug_hashes:
+        conn.execute(f"""
+            DELETE FROM review_statuses
```

Comment:
- Start a server with 6.18.2
- Analyze and store the following code to a run:
```cpp
int main() {
  sizeof(42);
}
```
- Analyze and store the following code to a run **different run**:
```cpp
int main() {
  // codechecker_false_positive [all] Test...
  sizeof(42);
}
```
- If you check the database it will contain the following information:
![image](https://user-images.githubusercontent.com/6695818/157409386-e4187d36-b014-4c77-9137-683aa097a856.png)
![image](https://user-images.githubusercontent.com/6695818/157409456-e7a87e90-a012-4093-a503-f178bdbe0525.png)
- Stop the server and start a server from your branch and do the migration.
- Now the database contains the following information:
![image](https://user-images.githubusercontent.com/6695818/157409597-db1e78f8-5a0f-4a73-b2f6-6ab71db5b969.png)
![image](https://user-images.githubusercontent.com/6695818/157409649-1871376e-abfe-4539-92dd-4defe6da5c5a.png)

Now the global review_status database is empty.This is an important change so it would be good to mention this in the PR's comment that after the migration what will happen. I will also insert this information in the release notes.

Reply:
Can you please update the PR message too? This is the one what the users will see when they visit your PR.